rust/parquet/src/arrow/array_reader.rs

```diff
@@ -653,11 +830,14 @@ impl<'a> ArrayReaderBuilder {
             PhysicalType::DOUBLE => Ok(Box::new(
                 PrimitiveArrayReader::<DoubleType>::new(page_iterator, column_desc)?,
             )),
-            PhysicalType::BYTE_ARRAY => Ok(Box::new(PrimitiveArrayReader::<
-                ByteArrayType,
-            >::new(
-                page_iterator, column_desc
-            )?)),
+            PhysicalType::BYTE_ARRAY
+                if cur_type.get_basic_info().logical_type() == LogicalType::UTF8 =>
+            {
+                Ok(Box::new(ComplexObjectArrayReader::<
```

Comment:
Out of curiousity, would the `ComplexObjectArrayReader` allow us to implement `arrow::BinaryArray` and `arrow::FixedSizeBinaryArray` in future?

Reply:
Yes, we just need converter for those types.